pkg/retry/options.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"math"
+)
+
+const (
+	// defaultBackoffBaseInMs is the initial duration, in Millisecond
+	defaultBackoffBaseInMs = 10.0
+	// defaultBackoffCapInMs is the max amount of duration, in Millisecond
+	defaultBackoffCapInMs = 100.0
+	defaultMaxTries       = 3
+)
+
+// Option ...
+type Option func(*retryOptions)
+
+// retryOptions ...
+type retryOptions struct {
+	maxTries    float64
+	backoffBase float64
+	backoffCap  float64
+	// isRetryable checks the error is safe to retry or not, eg. "context.Canceled" better not retry
+	isRetryable func(error) bool
+}
+
+func newRetryOptions() *retryOptions {
+	return &retryOptions{
+		maxTries:    defaultMaxTries,
+		backoffBase: defaultBackoffBaseInMs,
+		backoffCap:  defaultBackoffCapInMs,
+		isRetryable: func(err error) bool { return true },
+	}
+}
+
+// WithBackoffBaseDelay configures the initial delay
+func WithBackoffBaseDelay(delayInMs int64) Option {
```

Comment:
Can we use `time.Duration` to represent the `delayInMs` instead of `int64`?

Reply:
we could, but normally the delay should be covered at ms or s level, so just a int64 seems all right and more convenient  IMO